surefire-provider/src/main/java/org/arquillian/smart/testing/surefire/provider/TestStrategyApplier.java

```diff
@@ -44,15 +49,17 @@ TestsToRun apply(Configuration configuration) {
                 + "For details on how to configure it head over to http://bit.ly/st-config");
             return Collections.emptySet();
         }
-        final Set<TestSelection> selectedTests = new LinkedHashSet<>();
+        final Collection<TestSelection> selectedTests = new ArrayList<>();
         for (final String strategy : strategies) {
             final TestExecutionPlanner plannerForStrategy = testExecutionPlannerLoader.getPlannerForStrategy(strategy);
             selectedTests.addAll(plannerForStrategy.getTests());
         }
         logger.info("Applied strategies: %s", strategies);
         logger.info("Applied usage: [%s]", configuration.getMode().getName());
 
-        return selectedTests
+        final Collection<TestSelection> testSelections = filterMergeAndOrderTestSelection(selectedTests, strategies);
```

Comment:
Yeah, sure - it can stay as a collection - it was just the first thing that came to my mind when I saw it.
When I was reading the code, I saw the collection and started asking myself - OK, what kind of collection - ordered or unordered one? so, then I had to decipher the logic in the method and figure it out. 
This particular comment is just my opinion, nothing that I would request to change...